cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -84,7 +83,7 @@ private predicate parameterIsRedefined(Parameter p) {
 class FieldAddress extends Operand {
   FieldAddressInstruction fai;
 
-  FieldAddress() { fai = this.getDef() }
+  FieldAddress() { fai = this.getDef() and not Ssa::ignoreOperand(this) }
```

Comment:
Otherwise we end up generating dataflow nodes that aren't used anywhere: https://github.com/github/codeql/pull/15040/files#diff-e9e4c8dbaa54c9f16e1533c09bb66ff3ec456e747f98de139672cfa0412a463cR40-R42. Consider, for exapmle, this:
```cpp
void f(int*);

struct S { int x; };

void test() {
  S s;  
  f(&s.x);
}
```
the IR will look like:
```
r1 = &s.x;
r2 = call to f       : r1
m3 = WriteSideEffect : r1
...
```
And that `WriteSideEffect` isn't used for dataflow (and neither is its operand), so we shouldn't generate a `PostUpdateNode` for that `r1` operand occurring on the `WriteSideEffect`.

Reply:
Ah, it's because we now do `operand = any(FieldAddress fa).getObjectAddressOperand()` instead of having a node for the field address.

Reply:
Exactly, yeah.